File: app/src/main/java/com/nononsenseapps/feeder/ui/ReaderActivity.java
package com.nononsenseapps.feeder.ui;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBar;
import android.view.MenuItem;

import com.nononsenseapps.feeder.R;
import com.nononsenseapps.feeder.db.FeedItemSQL;
import com.nononsenseapps.feeder.views.DrawShadowFrameLayout;

import static com.nononsenseapps.feeder.ui.FeedFragmentKt.ARG_FEED_URL;

/**
 * Displays feed items suitable for consumption.
 */
public class ReaderActivity extends BaseActivity {

    private static final String TAG = "ReaderActivity";
    /**
     * Used to store the last screen title.
     */
    private CharSequence mTitle;
    private Fragment mFragment;
    private DrawShadowFrameLayout mDrawShadowFrameLayout;

    /**
     * Sets the extras in the intent suitable for opening the item in question.
     *
     * @param intent  to fill extras in
     * @param rssItem to read
     */
    public static void setRssExtras(Intent intent, FeedItemSQL rssItem) {
        intent.putExtra(ReaderFragmentKt.ARG_ID, rssItem.getId());
        intent.putExtra(ReaderFragmentKt.ARG_TITLE, rssItem.getTitle());
        intent.putExtra(ReaderFragmentKt.ARG_DESCRIPTION, rssItem.getDescription());
        intent.putExtra(ReaderFragmentKt.ARG_LINK, rssItem.getLink());
        intent.putExtra(ReaderFragmentKt.ARG_ENCLOSURE, rssItem.getEnclosurelink());
        intent.putExtra(ReaderFragmentKt.ARG_IMAGEURL, rssItem.getImageurl());
        intent.putExtra(ReaderFragmentKt.ARG_DATE, rssItem.getPubDateString());
        intent.putExtra(ReaderFragmentKt.ARG_AUTHOR, rssItem.getAuthor());
        intent.putExtra(ReaderFragmentKt.ARG_FEEDTITLE, rssItem.getFeedtitle());
        intent.putExtra(ARG_FEED_URL, rssItem.getFeedUrl());
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reader);
        initializeActionBar();
        ActionBar ab = getSupportActionBar();
        if (ab != null) {
            ab.setDisplayHomeAsUpEnabled(true);
            ab.setHomeButtonEnabled(true);
            ab.setDisplayShowTitleEnabled(false);
        }

        mTitle = getTitle();

        if (savedInstanceState == null) {
            mFragment = getFragment();
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.container, mFragment, "single_pane").commit();
        } else {
            mFragment = getSupportFragmentManager().findFragmentByTag("single_pane");
        }

        mDrawShadowFrameLayout =
                (DrawShadowFrameLayout) findViewById(R.id.main_content);

    }

    /**
     * Initializes a fragment based on intent information
     *
     * @return ReaderFragment
     */
    private Fragment getFragment() {
        Intent i = getIntent();
        Fragment fragment = new ReaderFragment();
        fragment.setArguments(i.getExtras());
        return fragment;
    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);

        registerHideableHeaderView(findViewById(R.id.headerbar));
        //registerHideableFooterView(mCheckButton);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();
        if (id == android.R.id.home && mShouldFinishBack) {
            // Handled by super, except animation
            boolean val = super.onOptionsItemSelected(item);
            // Only care about exit transition
            //overridePendingTransition(0, R.anim.contract_to_center);
            return val;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onActionBarAutoShowOrHide(boolean shown) {
        super.onActionBarAutoShowOrHide(shown);
        mDrawShadowFrameLayout.setShadowVisible(shown, shown);
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        if (mShouldFinishBack) {
            // Only care about exit transition
            //overridePendingTransition(0, R.anim.contract_to_center);
        }
    }
}

File: app/src/main/java/com/nononsenseapps/feeder/ui/ReaderActivity.kt
package com.nononsenseapps.feeder.ui

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.fragment.app.Fragment
import com.nononsenseapps.feeder.R
import com.nononsenseapps.feeder.db.room.FeedItemWithFeed
import com.nononsenseapps.feeder.model.PreviewItem
import com.nononsenseapps.feeder.views.DrawShadowFrameLayout

/**
 * Displays feed items suitable for consumption.
 */
class ReaderActivity : BaseActivity() {
    private var fragment: Fragment? = null
    private var mDrawShadowFrameLayout: DrawShadowFrameLayout? = null

    /**
     * Initializes a fragment based on intent information
     *
     * @return ReaderFragment
     */
    private fun fragmentFromIntent(): ReaderFragment {
            val i = intent
            val fragment = ReaderFragment()
            fragment.arguments = i.extras
            return fragment
        }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reader)
        initializeActionBar()
        val ab = supportActionBar
        if (ab != null) {
            ab.setDisplayHomeAsUpEnabled(true)
            ab.setHomeButtonEnabled(true)
            ab.setDisplayShowTitleEnabled(false)
        }

        fragment = if (savedInstanceState == null) {
            fragmentFromIntent().also {
                supportFragmentManager.beginTransaction()
                        .add(R.id.container, it, "single_pane").commit()
            }
        } else {
            supportFragmentManager.findFragmentByTag("single_pane")
        }

        mDrawShadowFrameLayout = findViewById(R.id.main_content)

    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)

        registerHideableHeaderView(findViewById<View>(R.id.headerbar))
        //registerHideableFooterView(mCheckButton);
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId
        return if (id == android.R.id.home && mShouldFinishBack) {
            // Handled by super, except animation
            // Only care about exit transition
            //overridePendingTransition(0, R.anim.contract_to_center);
            super.onOptionsItemSelected(item)
        } else super.onOptionsItemSelected(item)
    }

    override fun onActionBarAutoShowOrHide(shown: Boolean) {
        super.onActionBarAutoShowOrHide(shown)
        mDrawShadowFrameLayout?.setShadowVisible(shown, shown)
    }

    override fun onBackPressed() {
        super.onBackPressed()
        if (mShouldFinishBack) {
            // Only care about exit transition
            //overridePendingTransition(0, R.anim.contract_to_center);
        }
    }

    companion object {

        /**
         * Sets the extras in the intent suitable for opening the item in question.
         *
         * @param intent  to fill extras in
         * @param rssItem to read
         */
        fun setRssExtras(intent: Intent, rssItem: PreviewItem) {
            intent.putExtra(ARG_ID, rssItem.id)
            intent.putExtra(ARG_TITLE, rssItem.plainTitle)
            intent.putExtra(ARG_DESCRIPTION, rssItem.plainSnippet)
            intent.putExtra(ARG_LINK, rssItem.link)
            intent.putExtra(ARG_ENCLOSURE, rssItem.enclosureLink)
            intent.putExtra(ARG_IMAGEURL, rssItem.imageUrl)
            intent.putExtra(ARG_DATE, rssItem.pubDateString)
            intent.putExtra(ARG_AUTHOR, rssItem.author)
            intent.putExtra(ARG_FEEDTITLE, rssItem.feedTitle)
            intent.putExtra(ARG_FEED_URL, rssItem.feedUrl.toString())
        }

        fun setRssExtras(intent: Intent, rssItem: FeedItemWithFeed) {
            intent.putExtra(ARG_ID, rssItem.id)
            intent.putExtra(ARG_TITLE, rssItem.plainTitle)
            intent.putExtra(ARG_DESCRIPTION, rssItem.plainSnippet)
            intent.putExtra(ARG_LINK, rssItem.link)
            intent.putExtra(ARG_ENCLOSURE, rssItem.enclosureLink)
            intent.putExtra(ARG_IMAGEURL, rssItem.imageUrl)
            intent.putExtra(ARG_DATE, rssItem.pubDateString)
            intent.putExtra(ARG_AUTHOR, rssItem.author)
            intent.putExtra(ARG_FEEDTITLE, rssItem.feedTitle)
            intent.putExtra(ARG_FEED_URL, rssItem.feedUrl.toString())
        }
    }
}
